modules/core/src/main/java/org/apache/ignite/spi/tracing/TracingSpi.java

```diff
@@ -61,20 +43,20 @@ default SpiSpecificSpan create(@NotNull String name) {
      * 0 and 1 have special meaning here, 0 means never 1 means always. Default value is 0 (never).
      * @return Created span.
      */
-    @NotNull SpiSpecificSpan create(
+    @NotNull S create(
         @NotNull String name,
-        @Nullable SpiSpecificSpan parentSpan,
+        @Nullable S parentSpan,
         double samplingRate);
 
     /**
      * Serializes span to byte array to send context over network.
      *
      * @param span Span.
      */
-    byte[] serialize(@NotNull SpiSpecificSpan span);
+    byte[] serialize(@NotNull S span);
 
     /**
-     * @return type of tracing spi as {@link TracingSpiType} instance.
+     * @return type of tracing spi as byte.
```

Comment:
It seems to me, it would be nice to have a list of all known types of tracing SPIs somewhere. WDYT?